### PR TITLE
Add spinning wheel

### PR DESCRIPTION
**Pull request by mndys**
## Value Proposition (User Story)
As a **user**,
I would like to see a **spinning wheel** when I click the spinButton
so the process feels a little **realer and more fun**
## Acceptance Criteria (Edge Cases)
- [x] wheel starts spinning on button click
- [x] wheel stops spinning after a few seconds
- [x] prompt is revealed after wheel stops spinning

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -7,32 +7,38 @@ import prompts from '../data/prompts.json'
...
-  No more spins until you reset.`
+  const INITIAL_PROMPT = 'Spin to receive your first prompt.'
+  const LAST_PROMPT = `That's it. Reset to start again.`
 
   const [currentPrompt, setCurrentPrompt] = useState(
-    loadFromLocal('currentPrompt') ?? INITIALPROMPT
+    loadFromLocal('currentPrompt') ?? INITIAL_PROMPT
   )
   const [history, setHistory] = useState(loadFromLocal('promptHistory') ?? [])
+  const [mustSpin, setMustSpin] = useState(false)
```
Comment (AnnekeSchuette) [inline src/components/App.jsx:20]: I find the name of this variable a little bit confusing - does it stand for the state "is spinning"? Maybe "isSpinning" and "setIsSpinning" would be clearer. But this is just my humble opinion.

File: src/components/App.jsx [hunk 2]
```diff
@@ -61,20 +67,23 @@ function App() {
...
-      setCurrentPrompt(LASTPROMPT)
-      saveToLocal('currentPrompt', LASTPROMPT)
+      setCurrentPrompt(LAST_PROMPT)
+      saveToLocal('currentPrompt', LAST_PROMPT)
       setHistory([...history, currentPrompt])
       saveToLocal('promptHistory', [...history, currentPrompt])
+      setMustSpin(false)
     }
   }
 }
```
Comment (AnnekeSchuette) [inline src/components/App.jsx:89]: Within the code, it is easy to understand what happens under the respective conditions. You could consider outsourcing these functions for the sake of clarity inside the App component.

File: src/components/App.jsx [hunk 3]
```diff
@@ -92,10 +101,11 @@ const Grid = styled.div`
 const Main = styled.main`
   display: grid;
   justify-content: center;
-  grid-template-rows: 170px min-content auto;
+  grid-template-rows: 5em min-content auto;
   grid-template-columns: 1fr;
   padding: clamp(30px, 10%, 100px) clamp(15px, 5%, 50px);
   overflow-y: auto;
+  overflow-x: hidden;
 `
```
Comment (AnnekeSchuette) [inline src/components/App.jsx:109]: These two declaration could be merged into `overflow: hidden auto;`

File: src/components/Wheel.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+export default function WheelComponent({ winner, mustSpin, setMustSpin }) {
+  const backgroundColors = [
+    '#f94144',
+    '#f3722c',
+    '#f8961e',
+    '#f9c74f',
+    '#90be6d',
+    '#43aa8b',
+    '#577590',
+  ]
```
Comment (AnnekeSchuette) [inline src/components/Wheel.jsx:14]: I can not evaluate, if you plan to use these colors in other parts of your app later on, but you could consider to exchange them with css variables in order to make them easier to change/maintain.

**Review by AnnekeSchuette — COMMENTED** [review]
Very clean code for a complex feature! 🚀 I just added some little suggestions, which do not necessarily have to be implemented. Nice work - the result on vercel looks awesome by the way! ✨ ✌🏻

File: src/components/App.jsx (hunk 1 above)
Comment (mndys) [reply src/components/App.jsx:20]: The name really is confusing! But I can't think of one that accurately describes what it does in a short form 😅 As long as `mustSpin === true`, exactly one spin process can be triggered. `mustSpin` needs to become `false` and then revert to `true` for the wheel to spin again.

File: src/components/App.jsx (hunk 2 above)
Comment (mndys) [reply src/components/App.jsx:89]: Very good suggestions, thank you! I've put it on my to-do list 😊

File: src/components/App.jsx (hunk 3 above)
Comment (mndys) [reply src/components/App.jsx:109]: 💡 done!

File: src/components/Wheel.jsx (hunk 4 above)
Comment (mndys) [reply src/components/Wheel.jsx:14]: I don't know yet 🤔 probably not, but if I do, I'll use variables! At least I'm planning on doing so with the other colors as well 😅

**Pull request merged**